packages/react-components/src/icons/src/EmbeddedIcon.tsx

```diff
@@ -0,0 +1,38 @@
+import { Children, ReactElement, ReactNode } from "react";
+import { augmentElement, createSizeAdapter } from "../../shared";
+
+export interface EmbeddedIconProps {
+    /**
+     * An icon can vary in size.
+     */
+    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "inherit";
+    /**
+     * React children.
+     */
+    children: ReactNode;
+    /**
+     * Allow any extra props
+     */
+    [key: string]: any
+}
+
+export const embeddedIconSize = createSizeAdapter({
+    "2xs": "2xs",
```

Comment:
now that we export a `Size` we should use it here

Reply:
Size is a type union, not an enum. So if we provide a string that is not allowed in Size, typescript will put it in error.